usage/app/model/UsageGroup.scala

```diff
@@ -50,74 +46,143 @@ object UsageGroup {
-    // Generate unique UUID to track extract job
-    val uuid = java.util.UUID.randomUUID.toString
+  private def createUsagesLogging(usage: MediaUsage) = {
+    Logger.info(s"Built MediaUsage for ${usage.mediaId}")
 
-    Logger.info(s"Extracting images (job-${uuid}) from ${content.id}")
+    usage.digitalUsageMetadata.map(meta => {
+      Logger.info(s"MediaUsage for ${usage.mediaId}: ${Json.toJson(meta)}")
+    })
 
+    usage.printUsageMetadata.map(meta => {
+      Logger.info(s"MediaUsage for ${usage.mediaId}: ${Json.toJson(meta)}")
+    })
+    usage
+  }
+
+  private def isNewContent(content: Content, usageStatus: UsageStatus): Boolean = {
     val dateLimit = new DateTime(Config.usageDateLimit)
     val contentFirstPublished = LiveContentApi.getContentFirstPublished(content)
-    val isNew = usageStatus match {
+    usageStatus match {
       case _: PublishedUsageStatus => contentFirstPublished
         .map(_.isAfter(dateLimit)).getOrElse(false)
       case _ => true
     }
+  }
 
+  private def extractMediaAtoms(uuid: String, content: Content, usageStatus: UsageStatus, isReindex: Boolean) = {
+    val isNew = isNewContent(content, usageStatus)
     val shouldRecordUsages = isNew || isReindex
 
-    def groupImageElements: Option[Seq[Element]] = content.elements.map(elements => {
-      elements.filter(_.`type` == ElementType.Image)
-        .groupBy(_.id)
-        .map(_._2.head).to[collection.immutable.Seq]
-    })
+    if (shouldRecordUsages) {
+      Logger.info(s"Passed shouldRecordUsages for media atom (job-${uuid})")
+      val groupedMediaAtoms = groupMediaAtoms(content)
+
+      if (groupedMediaAtoms.isEmpty) {
+        Logger.info(s"No Matching media atoms found (job-${uuid})")
+      } else {
+        Logger.info(s"${groupedMediaAtoms.length} media atoms found (job-${uuid})")
+        groupedMediaAtoms.map(atom => Logger.info(s"Matching media atom ${atom.id} found (job-${uuid})"))
+      }
+
+      groupedMediaAtoms
+    } else {
+      Logger.info(s"Failed shouldRecordUsages for media atoms: isNew-${isNew} isReindex-${isReindex} (job-${uuid})")
+      Seq.empty
+    }
+  }
+
+  private def groupMediaAtoms(content: Content) = {
+    val mediaAtoms = content.atoms match {
+      case Some(atoms) => {
+        atoms.media match {
+          case Some(mediaAtoms) => filterOutAtomsWithNoImage(mediaAtoms)
+          case _ => Seq.empty
+        }
+      }
+      case _ => Seq.empty
+    }
+    mediaAtoms
+  }
+
+  private def filterOutAtomsWithNoImage(atoms: Seq[Atom]): Seq[Atom] = {
+    for {
+      atom <- atoms
+      atomId <- getImageId(atom)
+      if atomId != None
+    } yield atom
+  }
+
+  private def getImageId(atom: Atom): Option[String] = {
+    try {
+      val posterImage = atom.data.asInstanceOf[AtomData.Media].media.posterImage
+      posterImage match {
+        case Some(image) => Some(image.mediaId.replace("https://api.media.gutools.co.uk/images/", ""))
```

Comment:
Not sure... Definitely nicer to pull it out but because the media atom atom maker always uses PROD grid even in CODE I'm not sure if this is the right value to use?